egs/fisher_callhome_spanish_st/st1/local/callhome_data_prep.sh

```diff
@@ -0,0 +1,166 @@
+#!/bin/bash
+#
+# Copyright 2014  Gaurav Kumar.   Apache 2.0
+# The input is the Callhome Spanish Dataset. (*.sph files)
+# In addition the transcripts are needed as well.
+# To be run from one directory above this script.
+
+# Note: when creating your own data preparation scripts, it's a good idea
+# to make sure that the speaker id (if present) is a prefix of the utterance
+# id, that the output scp file is sorted on utterance id, and that the
+# transcription file is exactly the same length as the scp file and is also
+# sorted on utterance id (missing transcriptions should be removed from the
+# scp file using e.g. scripts/filter_scp.pl)
+
+stage=0
+
+export LC_ALL=C
+
+
+if [ $# -lt 2 ]; then
+   echo "Arguments should be the location of the Callhome Spanish Speech and Transcript Directories, se
+e ../run.sh for example."
+   exit 1;
+fi
+
+cdir=`pwd`
+dir=`pwd`/data/local/data
+local=`pwd`/local
+utils=`pwd`/utils
+tmpdir=`pwd`/data/local/tmp
+
+. ./path.sh || exit 1; # Needed for KALDI_ROOT
+export PATH=$PATH:$KALDI_ROOT/tools/irstlm/bin
+sph2pipe=$KALDI_ROOT/tools/sph2pipe_v2.5/sph2pipe
+if [ ! -x $sph2pipe ]; then
+   echo "Could not find (or execute) the sph2pipe program at $sph2pipe";
+   exit 1;
+fi
+cd $dir
+
+# Make directory of links to the WSJ disks such as 11-13.1.  This relies on the command
+# line arguments being absolute pathnames.
+#rm -r links/ 2>/dev/null
+mkdir -p links/
+ln -s $* links
+
+# Basic spot checks to see if we got the data that we needed
+if [ ! -d links/LDC96S35 -o ! -d links/LDC96T17 ];
+then
+        echo "The speech and the data directories need to be named LDC96S35 and LDC96T17 respecti
+vely"
+        exit 1;
+fi
+
+if [ ! -d links/LDC96S35/CALLHOME/SPANISH/SPEECH/DEVTEST -o ! -d links/LDC96S35/CALLHOME/SPANISH/SPEECH/EVLTEST -o ! -d links/LDC96S35/CALLHOME/SPANISH/SPEECH/TRAIN ];
```

Comment:
When I downloaded LDC96S35 from LDC yesterday, the path looks like `LDC96S35/callhome/spanish/speech/devtest`

how about this?
```bash
if [ -d links/LDC96S35/CALLHOME/SPANISH/SPEECH/DEVTEST -o -d links/LDC96S35/CALLHOME/SPANISH/SPEECH/EVLTEST -o -d links/LDC96S35/CALLHOME/SPANISH/SPEECH/TRAIN ]
then
    speech_train=$dir/links/LDC96S35/CALLHOME/SPANISH/SPEECH/TRAIN
    speech_dev=$dir/links/LDC96S35/CALLHOME/SPANISH/SPEECH/DEVTEST
    speech_test=$dir/links/LDC96S35/CALLHOME/SPANISH/SPEECH/EVLTEST
    fcount_train=`find ${speech_train} -iname '*.SPH' | wc -l`
    fcount_dev=`find ${speech_dev} -iname '*.SPH' | wc -l`
    fcount_test=`find ${speech_test} -iname '*.SPH' | wc -l`
elif [ -d links/LDC96S35/callhome/spanish/speech/devtest -o -d links/LDC96S35/callhome/spanish/speech/evltest -o -d links/LDC96S35/callhome/spanish/speech/train ]
then
    speech_train=$dir/links/LDC96S35/callhome/spanish/speech/train
    speech_dev=$dir/links/LDC96S35/callhome/spanish/speech/devtest
    speech_test=$dir/links/LDC96S35/callhome/spanish/speech/evltest
    fcount_train=`find ${speech_train} -iname '*.sph' | wc -l`
    fcount_dev=`find ${speech_dev} -iname '*.sph' | wc -l`
    fcount_test=`find ${speech_test} -iname '*.sph' | wc -l`
else
        echo "Dev, Eval or Train directories missing or not properly organised within the speech data dir"
        exit 1;
fi
```